webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +87,1137 @@ public WebSiteSessionController createComponentSessionController(
-        String topicId = request.getParameter("Id");
+          request.setAttribute("Site", descriptionSite);
+          request.setAttribute("AllIcons", scc.getAllIcons());
+          request.setAttribute("ListeIcones", listeIcones);
+          request.setAttribute("UploadOk", Boolean.TRUE);
+          request.setAttribute("SearchOk", Boolean.FALSE);
+          destination = "/webSites/jsp/descUpload.jsp";
+          break;
 
-        scc.changeTopicsOrder("down", topicId);
+        case -2:
+          request.setAttribute("Site", descriptionSite);
+          request.setAttribute("AllIcons", scc.getAllIcons());
+          request.setAttribute("ListeIcones", listeIcones);
+          request.setAttribute("UploadOk", Boolean.FALSE);
+          destination = "/webSites/jsp/descUpload.jsp";
+          break;
 
-        String id = scc.getSessionTopic().getNodePK().getId();
-        FolderDetail webSitesCurrentFolder = scc.getFolder(id);
-        scc.setSessionTopic(webSitesCurrentFolder);
-        request.setAttribute("CurrentFolder", webSitesCurrentFolder);
+        case -3: // the zip content isn't correct
+          scc.deleteDirectory(scc.getWebSitePathById(descriptionSite.getId()));
+          request.setAttribute("Site", descriptionSite);
+          request.setAttribute("AllIcons", scc.getAllIcons());
+          request.setAttribute("ListeIcones", listeIcones);
+          request.setAttribute("UploadOk", Boolean.FALSE);
+          destination = "/webSites/jsp/descUpload.jsp";
+          break;
+      }
+    }
+    return destination;
+  }
 
-        destination = "/webSites/jsp/organize.jsp?Action=Search&Id=" + id;
-      } else if ("PubUp".equals(function)) {
-        String pubId = request.getParameter("Id");
+  private String uploadFile(final WebSiteSessionController scc, final HttpRequest request,
+      final String userRole) throws Exception {
 
-        scc.changePubsOrder(pubId, -1);
+    List<FileItem> items = request.getFileItems();
+    boolean status = true;
+    String thePath = FileUploadUtil.getParameter(items, "path");
+    scc.checkPath(thePath);
+    FileItem item = FileUploadUtil.getFile(items);
+    if (item != null) {
+      status = (scc.addFileIntoWebSite(thePath, item) == 0);
+    }
 
-        String id = scc.getSessionTopic().getNodePK().getId();
-        FolderDetail webSitesCurrentFolder = scc.getFolder(id);
-        scc.setSessionTopic(webSitesCurrentFolder);
-        request.setAttribute("CurrentFolder", webSitesCurrentFolder);
+    request.setAttribute("UploadOk", status);
 
-        destination = "/webSites/jsp/organize.jsp?Action=Search&Id=" + id;
-      } else if ("PubDown".equals(function)) {
-        String pubId = request.getParameter("Id");
+    return  "/webSites/jsp/uploadFile.jsp?path=" + thePath;
+  }
 
-        scc.changePubsOrder(pubId, 1);
+  private String designSite(final WebSiteSessionController scc, final HttpRequest request,
+      final String userRole) throws WebSitesException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "userRole". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1172)